src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/page/CommentTabPanel.kt

```diff
@@ -38,7 +38,7 @@ class CommentTabPanel(
         val comments = driver.findElements(By.cssSelector("#issue_actions_container > div.activity-comment"))
         return comments.map {
             val id = it.getAttribute("id").split("-")[1]
-            val url = it.findElement(By.cssSelector(".comment-created-date-link")).getAttribute("href")
+            val url = it.findElement(By.cssSelector(""".action-head a[href^="/browse/"]""")).getAttribute("href")
```

Comment:
I'm trying to finish this PR after a while.

Taking a look again at this change I'm not sure if this selector change is moving the code in the right direction. The previous selector didn't work with older Jira versions, however was probably more "semantic". New selector has an assumption about contract which will unlikely change, but it's still some knowledge which probably shouldn't be here.

I'm thining if it wouldn't be better to somehow query this element both ways and select the first match. Not sure how to do that without compromising the readability and noise.

CC: @dagguh

Reply:
@pczuj Maybe something like this? 
```
val jiraVersionsPriorTo817Selector = ".comment-created-date-link"
val jiraVersionsSelectorAfter817 = """.action-head a[href^="/browse/"]"""
val url = it.findElement(By.cssSelector("$jiraVersionsPriorTo817Selector, $jiraVersionsSelectorAfter817")).getAttribute("href")
```

Reply:
Another way to "or":
```kotlin
val url = it.findElements(By.cssSelector(".comment-created-date-link"))
    .plus(it.findElements(By.cssSelector(".action-head a[href^='/browse/']")))
    .first()
    .getAttribute("href")
```
But I like @mgrzaslewicz's version, because it explains the 8.1.7 reasoning.

Reply:
There's some semantic misalignment.

The `.action-head` - "action" is old Jira wording for comment, so that's okayish, but "head" means the comment header. Header is a location, not a meaning. We're interested in the comment url/link, wherever it is located.

The `href^="/browse/"` assumes the structure of `Comment.url`, but this prefix is not comment-specific. If we need to assume the `href`, we should go for `*#comment*` or `*com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel*`.

The old `.comment-created-date-link` was quite good semantically, it's about a "comment link". It's unnecessarily polluted by "created date", but we can live with that.

Could you post a HTML snippet which doesn't work with `.comment-created-date-link`? It could be the case that the `.action-head` + href is the lesser of all evils, but let's see.

Reply:
In Jira 8.13.0 rendered comment HTML looks like this:
```
<div id="comment-10001" class="issue-data-block activity-comment twixi-block  expanded">
    <div class="twixi-wrap verbose actionContainer">
        <div class="action-head">
            <a href="#" title="Collapse comment" class="twixi"><span class="icon-default aui-icon aui-icon-small aui-iconfont-expanded"><span>Hide</span></span></a>
            <div class="action-links">
                <a href="/browse/FIR-10?focusedCommentId=10001&amp;page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-10001" title="Right click and copy link for a permanent link to this comment." class="activitymodule-link issue-comment-action">
                    <span class="icon-default aui-icon aui-icon-small aui-iconfont-link">Permalink</span></a>
                <a id="edit_comment_10001" href="/secure/EditComment!default.jspa?id=10009&amp;commentId=10001" title="Edit" class="edit-comment issue-comment-action">
                    <span class="icon-default aui-icon aui-icon-small aui-iconfont-edit">Edit</span></a>
                <a id="delete_comment_10001" href="/secure/DeleteComment!default.jspa?id=10009&amp;commentId=10001" title="Delete" class="delete-comment issue-comment-action">
                    <span class="icon-default aui-icon aui-icon-small aui-iconfont-delete">Delete</span></a>
            </div>
            <div class="action-details">





                <a class="user-hover user-avatar" rel="admin" id="commentauthor_10001_verbose" href="/secure/ViewProfile.jspa?name=admin"><span class="aui-avatar aui-avatar-xsmall"><span class="aui-avatar-inner"><img src="https://www.gravatar.com/avatar/e64c7d89f26bd1972efa854d13d7dd61?d=mm&amp;s=16" alt="admin"></span></span> admin</a>
                added a comment  - <span class="commentdate_10001_verbose subText"><span class="date user-tz" title="07/Oct/22 7:37 AM"><time class="livestamp" datetime="2022-10-07T07:37:41+0200">1 week ago</time></span></span>  </div>
        </div>
        <div class="action-body flooded"><p>Joined Sample Sprint 2 7 days 9 hours 10 minutes ago</p> </div>
    </div>
    <div class="twixi-wrap concise actionContainer">
        <div class="action-head">
            <a href="#" title="Expand comment" class="twixi"><span class="icon-default aui-icon aui-icon-small aui-iconfont-collapsed"><span>Show</span></span></a>
            <div class="action-details flooded">






                <a class="user-hover user-avatar" rel="admin" id="commentauthor_10001_concise" href="/secure/ViewProfile.jspa?name=admin"><span class="aui-avatar aui-avatar-xsmall"><span class="aui-avatar-inner"><img src="https://www.gravatar.com/avatar/e64c7d89f26bd1972efa854d13d7dd61?d=mm&amp;s=16" alt="admin"></span></span> admin</a>
                added a comment  - <span class="commentdate_10001_concise subText"><span class="date user-tz" title="07/Oct/22 7:37 AM"><time class="livestamp" datetime="2022-10-07T07:37:41+0200">1 week ago</time></span></span>                    Joined Sample Sprint 2 7 days 9 hours 10 minutes ago              </div>
        </div>
    </div>
</div>
```